repository_service_tuf_worker/repository.py

```diff
@@ -1108,6 +1112,110 @@ def bump_online_roles(self, force: Optional[bool] = False) -> bool:
 
         return True
 
+    def _run_force_online_metadata_update(
+        self, payload: List[str]
+    ) -> List[str]:
+        "Run the actual metadata update for set of online roles."
+        roles_diff: List[str] = []
+        if Targets.type in payload and "bins" in payload:
+            self._run_online_roles_bump(force=True)
+            roles_diff = [Targets.type, "bins", Snapshot.type, Timestamp.type]
+
+        elif Targets.type in payload or "bins" in payload:
+            if Targets.type in payload:
+                targets = self._storage_backend.get(Targets.type)
+                self._bump_and_persist(targets, Targets.type)
+                logging.info("Bumped version of 'Targets' role")
+                self.bump_snapshot(force=True)
+                roles_diff = [Targets.type, Snapshot.type, Timestamp.type]
+            else:
+                # Bump all bins
+                self._update_timestamp(self._update_snapshot(bump_all=True))
+                roles_diff = ["bins", Snapshot.type, Timestamp.type]
+
+        elif Snapshot.type in payload:
+            self.bump_snapshot(force=True)
+            roles_diff = [Snapshot.type, Timestamp.type]
+
+        elif Timestamp.type in payload:
+            snapshot = self._storage_backend.get(Snapshot.type)
+            self._update_timestamp(snapshot.signed.version)
+            roles_diff = [Timestamp.type]
+
+        return roles_diff
+
+    def force_online_metadata_update(
+        self,
+        payload: Dict[str, Any],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        """Force metadata update on given online roles."""
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+        if bootstrap is None or "pre-" in bootstrap or "signing-" in bootstrap:
+            return self._task_result(
+                TaskName.FORCE_ONLINE_METADATA_UPDATE,
+                False,
+                {
+                    "message": "Force new online metadata update failed",
+                    "error": "New metadata updates requre completed bootstrap",
+                },
+            )
+        if len(payload["roles"]) < 1:
+            return self._task_result(
+                TaskName.FORCE_ONLINE_METADATA_UPDATE,
+                True,
+                {
+                    "message": "No online metadata roles given",
+                },
+            )
+
```

Comment:
In the API, [we allow users to submit offline roles without checking it (example: Root)](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/5594c1684064705c12f660ed68858c82636ba154/repository_service_tuf_api/metadata.py)


In the Worker, we also don't verify it. It will return a task SUCCESS with status `True`.
I suggest we verify if the role

Note:  We can verify Root and Targets now. We have in the settings/redis [a setting if the `Targets` is offline or online key](https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/0463b088095b0f8a98be91b9f80591f7d8ffa088/repository_service_tuf_worker/repository.py#L493-L495) -- for future feature that allows offline targets.

Reply:
I changed it and now we only allow online roles and specifically check if targets is an online role.